plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AWSContainerOptionsMapper.groovy

```diff
@@ -0,0 +1,170 @@
+package nextflow.cloud.aws.batch
+
+import com.amazonaws.services.batch.model.ContainerProperties
+import com.amazonaws.services.batch.model.KeyValuePair
+import com.amazonaws.services.batch.model.LinuxParameters
+import com.amazonaws.services.batch.model.Tmpfs
+import com.amazonaws.services.batch.model.Ulimit
+
+/**
+ * Maps task container options to AWS container properties
+ *
+ * @see <a href="https://docs.docker.com/engine/reference/commandline/run/">Docker run</a>
+ * @see <a href="https://docs.aws.amazon.com/batch/latest/APIReference/API_ContainerProperties.html">API Container Properties</a>
+ * @author Manuele Simi <manuele.simi@gmail.com>
+ */
+class AWSContainerOptionsMapper {
+
+    def options
+
+    protected AWSContainerOptionsMapper(String containerOptions) {
+        options = containerOptions.trim().split("\\s+")
```

Comment:
Too weak, the container option values could container blanks and quotes. Also I think the parsing should not be done by this component, which could take instead a Map object

Reply:
That makes sense. I also agree that the parsing should occur earlier. Let me think about it.

Reply:
There's an implementation in the commit I've linked above

Reply:
I saw it :)

Reply:
I saw it, thanks :)